src/Native/System.Native/pal_mount.cpp

```diff
@@ -134,7 +141,7 @@ SystemNative_GetFormatInfoForMountPoint(const char* name, char* formatNameBuffer
         }
 #else
         assert(formatType != nullptr);
-        *formatType = stats.f_type;
+        *formatType = GetFTypeAsInt64(stats.f_type);
```

Comment:
I don't see how this is any different from just casting to int64 inline and unconditionally. Inline the method call manually to see what I mean.

If the reason why an implicit conversion to int64 doesn't always work is that the value is sometimes unsigned, then you can use our SignedCast helper instead of static_cast which will at least assert if this cast would overflow.